src/libcore/alloc.rs

```diff
@@ -213,18 +213,19 @@ impl Layout {
     /// Creates a layout by rounding the size of this layout up to a multiple
     /// of the layout's alignment.
     ///
-    /// Returns `Err` if the padded size would overflow.
-    ///
     /// This is equivalent to adding the result of `padding_needed_for`
     /// to the layout's current size.
     #[unstable(feature = "alloc_layout_extra", issue = "55724")]
     #[inline]
-    pub fn pad_to_align(&self) -> Result<Layout, LayoutErr> {
+    pub fn pad_to_align(&self) -> Layout {
         let pad = self.padding_needed_for(self.align());
-        let new_size = self.size().checked_add(pad)
-            .ok_or(LayoutErr { private: () })?;
+        // This cannot overflow: it is an invariant of Layout that
+        // > `size`, when rounded up to the nearest multiple of `align`,
+        // > must not overflow (i.e., the rounded value must be less than
```

Comment:
"less than"? not "less or equal to"?

Reply:
This is just quoted from `from_size_align`, though I agree lte is the correct bound here:

https://github.com/rust-lang/rust/blob/6773064b0529c258c4f69bb7848371553447b088/src/libcore/alloc.rs#L55-L94

Reply:
Looks like this goes back all the way to https://github.com/rust-lang/rust/pull/42313.... @pnkfelix @alexcrichton  do you remember why you made this "less than", not "less or equal"?

Reply:
`usize::MAX` isn't a multiple of two so if you round up to the nearest multiple of `align` you're by definition not equal to `usize::MAX` nor `usize::MAX - 1`, so I don't think that exact phrasing matters here. If I'm wrong then no, I have no recollection of why this one phrase is buried in documentation landed 2 years ago.